### DISCUSSION TITLE
Seeed Xiao ESP32C3 USB serial port

### DISCUSSION THREAD

**Discussion by Shuptuu** [someweisguy/esp_dmx#145] [Q&A] [answered]
Hi, 
Is there a way to use the USB serial port on a Seeed Esp32C3 board as the dmx port?
If I set the dmxport to 0, then what are the pins to be set for transmit and receive?
I'd like to get dmx packets from my computer through the usb cable and send them using ESP-NOW. 
Bonus: forward thoses dmx packets on an other serial port.
Any way to do it?
Would it be possible to do it with a Seeed Xiao ESP32C3? or with the ESP32S3 version?

**Comment by someweisguy** [comment, marked as answer]
Thanks for reaching out! This should be possible, though I haven't tried with this specific hardware. It appears that the TX and RX pins for UART0 are 21 and 20 by default. You would just need one more pin to use as the enable pin. It appears that all the other available pins on the Seeed board would work - perhaps you could use GPIO7.

One thing you would have to ensure is that UART0 is not being used by the ESP32 for any other purpose. This can be particularly tricky because it is how messages are logged to the Serial Monitor. 

To forward DMX packets from one port to another, you would need to install two DMX drivers. One DMX driver would receive the DMX data and the second would be the forwarding port. In essence, you could use `dmx_read()` to read from port 0, and then write that data to port 1 using `dmx_write()` and `dmx_send()`.

Hope this helps! Let me know if you have any other questions!

**Reply by Shuptuu** [reply]
Hi,
Thanks for your answer.
Pins 21 & 20 are for UART1, not 0 and the problem is that on this Esp32c3, the USB differential data pins (D+ & D-) are going directly to the ESP and managed internally so there's no tx and rx gpio for uart0 (or I didn't find them...). Is there a way to say to the driver to keep its original pinout? Like a UART_PIN_NO_CHANGE value we can pass to the dmx _set_pin function?

**Reply by Shuptuu** [reply]
To be more precise: on esp32c3, there's one serial over usb and 2 hardware uarts. You're right, pin 21 and 20 are for uart0, but this is not the serial over usb.
Found this from espressif: https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-guides/usb-serial-jtag-console.html
Looks like it's gonna be difficult to use it for dmx...
